### PR TITLE
Faster routine

### PR DESCRIPTION
**Pull request by Pierlou**
- faster routine, now almost only dependant on file size
- add possibility to see logs with verbose=True
- next steps: change test method to prevent all tests to run on all columns, maybe switch to polars (if substantially faster)

### CODE REVIEW

**Comment by Pierlou** [comment]
Don't pay attention to first 3 commits, wrong branch 🙃

File: csv_detective/detect_fields/FR/geo/code_commune_insee/__init__.py [hunk 1]
```diff
@@ -4,6 +4,9 @@
 PROPORTION = 0.75
 f = open(join(dirname(__file__), 'code_commune_insee.txt'), 'r')
 codes_insee = f.read().split('\n')
+# removing empty str du to additionnal line in file
```
Comment (geoffreyaldebert) [inline csv_detective/detect_fields/FR/geo/code_commune_insee/__init__.py:7]: are we forced to do this deletion. Could you not just delete row in file ?

File: csv_detective/detect_fields/FR/geo/commune/__init__.py [hunk 2]
```diff
@@ -4,6 +4,9 @@
 PROPORTION = 0.9
 f = open(join(dirname(__file__), 'commune.txt'), 'r')
 codes_commune = f.read().split('\n')
+# removing empty str du to additionnal line in file
```
Comment (geoffreyaldebert) [inline csv_detective/detect_fields/FR/geo/commune/__init__.py:7]: same remark

File: csv_detective/detect_fields/FR/geo/departement/__init__.py [hunk 3]
```diff
@@ -4,6 +4,9 @@
 PROPORTION = 0.9
 f = open(join(dirname(__file__), 'departement.txt'), 'r')
 codes_departement = f.read().split('\n')
+# removing empty str du to additionnal line in file
```
Comment (geoffreyaldebert) [inline csv_detective/detect_fields/FR/geo/departement/__init__.py:7]: same

File: csv_detective/detect_fields/FR/geo/insee_canton/__init__.py [hunk 4]
```diff
@@ -5,6 +5,9 @@
 PROPORTION = 0.9
 f = open(join(dirname(__file__), 'cantons.txt'), 'r')
 cantons = f.read().split('\n')
+# removing empty str du to additionnal line in file
```
Comment (geoffreyaldebert) [inline csv_detective/detect_fields/FR/geo/insee_canton/__init__.py:8]: same

File: csv_detective/detect_fields/FR/geo/region/__init__.py [hunk 5]
```diff
@@ -2,12 +2,15 @@
 from csv_detective.process_text import _process_text
 
 PROPORTION = 1
+f = open(join(dirname(__file__), 'region.txt'), 'r')
+regions = f.read().split('\n')
+# removing empty str du to additionnal line in file
+del regions[-1]
```
Comment (geoffreyaldebert) [inline csv_detective/detect_fields/FR/geo/region/__init__.py:8]: same

File: csv_detective/detect_fields/FR/other/csp_insee/__init__.py [hunk 6]
```diff
@@ -4,6 +4,9 @@
 PROPORTION = 1
 f = open(join(dirname(__file__), 'csp_insee.txt'), 'r')
 codes_insee = f.read().split('\n')
+# removing empty str du to additionnal line in file
+del codes_insee[-1]
```
Comment (geoffreyaldebert) [inline csv_detective/detect_fields/FR/other/csp_insee/__init__.py:8]: same

File: csv_detective/detect_fields/FR/other/insee_ape700/__init__.py [hunk 7]
```diff
@@ -4,6 +4,9 @@
 PROPORTION = 1
 f = open(join(dirname(__file__), 'insee_ape700.txt'), 'r')
 condes_insee_ape = f.read().split('\n')
+# removing empty str du to additionnal line in file
+del condes_insee_ape[-1]
```
Comment (geoffreyaldebert) [inline csv_detective/detect_fields/FR/other/insee_ape700/__init__.py:8]: same

File: csv_detective/detection.py [hunk 8]
```diff
@@ -2,9 +2,14 @@
 import numpy as np
 from cchardet import UniversalDetector
 from ast import literal_eval
+import logging
+from time import time
+from csv_detective.utils import print_colored_logs
```
Comment (geoffreyaldebert) [inline csv_detective/detection.py:7]: rename function and its call to something like `display_logs_depending_process_time`

File: csv_detective/detection.py [hunk 9]
```diff
@@ -2,9 +2,14 @@
...
 
 
-def detect_continuous_variable(table, continuous_th=0.9):
+logging.basicConfig(level=logging.INFO)
```
Comment (geoffreyaldebert) [inline csv_detective/detection.py:10]: can you variabilize this. With a `LOG_LEVEL` variable for instance. With a default to `logging.INFO`if not set.

**Review by geoffreyaldebert — APPROVED** [review]

File: csv_detective/detection.py (hunk 9 above)
Comment (Pierlou) [reply csv_detective/detection.py:10]: This is set in three different files, I can set it in one file only and import it in the others. Is this what you mean?

**Pull request merged**